src/main/java/org/opensearch/securityanalytics/util/DetectorFactory.java

```diff
@@ -0,0 +1,50 @@
+package org.opensearch.securityanalytics.util;
+
+import org.opensearch.commons.alerting.model.IntervalSchedule;
+import org.opensearch.securityanalytics.model.Detector;
+import org.opensearch.securityanalytics.model.DetectorInput;
+import org.opensearch.securityanalytics.model.DetectorRule;
+
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DetectorFactory {
+    /* Creates a Detector object with the given rules, log type, and index name */
+    public static Detector createDetector(String integration, List<String> detectorRules) {
+
+        List<DetectorRule> rules = new ArrayList<>();
+        detectorRules.forEach(rule -> rules.add(new DetectorRule(rule)));
+
+        Long version = 1L;
+        String name = integration + "-detector";
+        String description = "Detector for " + integration + " integration";
+        String dataStream = "wazuh-events-v5-other";
```

Comment:
We need to track the correct index/data stream that belongs to the integration's category.